en/docs/architecture-diagram/architecture-diagram.md

```diff
@@ -0,0 +1,126 @@
+
+```ballerina
+@display {
+   label: "CurrencyService",
+   id: "currency"
+}
+@grpc:ServiceDescriptor {descriptor: ROOT_DESCRIPTOR, descMap: getDescriptorMapDemo()}
+service "CurrencyService" on ep {final map<decimal> & readonly currencyMap;
+   function init() returns error? {
+      json currencyJson = check io:fileReadJson(currencyJsonPath);
+      self.currencyMap = check parseCurrencyJson(currencyJson).cloneReadOnly();
+   }
+
+   remote function GetSupportedCurrencies(Empty value) returns GetSupportedCurrenciesResponse|error {
+      return {currency_codes: self.currencyMap.keys()};
+   }
+}
+```
+
+If other services interact with this service, this same identifier has to be provided upon creating the client endpoint. By doing so, it makes it possible to uniquely identify and link the service interactions between one another. 
+
+For example, the code snippet below invokes the `Currency Service` from a different component.
+
+```ballerina
+isolated function getSupportedCurrencies() returns string[]|error {
+   @display {
+      label: "CurrencyService",
+   id: "currency"
+   }
+   final CurrencyServiceClient currencyClient = check new ("http://localhost:9093");
+   GetSupportedCurrenciesResponse|grpc:Error supportedCurrencies = currencyClient->GetSupportedCurrencies({});
+   if supportedCurrencies is grpc:Error {
+      log:printError("failed to call getSupportedCurrencies from currency service");
+      return supportedCurrencies;
+   }
+   return supportedCurrencies.currency_codes;
+}
+```
+
+### Invoke service resources
+
+The resource invocations between services need to be done using [client access actions](https://ballerina.io/downloads/swan-lake-release-notes/swan-lake-2201.2.0#support-for-resource-methods-in-client-objects).
+
+## Explore the `Project Design Diagram` features
+
+The [Google Cloud Platform demo](https://github.com/ballerina-guides/gcp-microservices-demo/tree/project-design-demo) is used to demonstrate the different capabilities of the diagram. This sample has 10 services (9 GRPC and one HTTP) in 10 separate Ballerina packages. The `ui folder` is not a Ballerina package, hence would not make an impact.
+
+As per the [Related concepts](#a-project-workspace) section, each of these packages needs to be loaded into a VS Code workspace. This could be done using any of the following approaches.
+1. Use the VS Code workspace file that is available in the repository
+2. Add each Ballerina package to the workspace using the `Add folder to workspace` feature available in VS Code
+
+If the workspace has been set up accurately, you would be able to see the following folder structure in the explorer.
+>**Note:** The workspace name would be different if you set up the workspace from scratch.
+
+![Workspace Structure](images/workspace-structure.png)
```

Comment:
Can we move all the images to the docs/img folder [1] and update the links please?

[1] https://github.com/wso2/docs-ballerina-visual-studio-code-plugin/tree/main/en/docs/img